### PR TITLE
fix(mattermost): use correct envs, fix prod

### LINKED ISSUE #127: bug(mattermost): missing env variable
\from @revolunet  https://github.com/SocialGouv/gitlab-ci-yml/pull/125#issuecomment-611486434

> sniff il y a un bug pour detecter l'environnement, visiblement il manque les infos relatives à l'environnement 
testé avec https://github.com/SocialGouv/gitlab-ci-yml/blob/revolunet-patch-1/base_notify_github_stage.yml sans succes :/

### PR DESCRIPTION
**Pull request by revolunet**
fix #127 and #128 thanks @douglasduteil

### CODE REVIEW

File: base_notify_mattermost.yml [hunk 1]
```diff
@@ -9,8 +8,27 @@
...
+  only:
+    refs:
+      - master
+  environment:
+    name: prod
+  variables:
+    MATTERMOST_CHANNEL: ${MATTERMOST_CHANNEL:-gitlab/notifications}
   before_script:
-    - HOST="https://${CI_ENVIRONMENT_SLUG}-${CI_PROJECT_NAME}.${KUBE_INGRESS_BASE_DOMAIN}"
+    - HOST="https://${CI_PROJECT_NAME}.${KUBE_INGRESS_BASE_DOMAIN}"
```
Comment (douglasduteil) [inline base_notify_mattermost.yml:31]: @revolunet This is `autodevops_simple_app` centric as it's using the `autodevops_simple_app` env name, urls and only/except 
I think we should put this outside of the `notify_mattermost` (in autodevops like the github notify and/or the documentation) or prefix it with the env/only/expect goal

File: base_notify_mattermost.yml [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
 
+.resolve_deploy_environment: &resolve_deploy_environment |
+  set -x
+
+  if [[ -n "${PRODUCTION+x}" ]]; then
+    export ENVIRONMENT="production"
+  else
+    export ENVIRONMENT=${ENVIRONMENT:-"staging"}
+  fi
+
```
Comment (douglasduteil) [inline base_notify_mattermost.yml:13]: Why do you need `$ENVIRONMENT` for ?

Comment (revolunet) [reply base_notify_mattermost.yml]: removed !

**Comment by revolunet** [comment]
up :)

**Pull request closed without merging**